src/main/java/seedu/address/model/pet/comparator/PetDropOffTimeComparator.java

```diff
@@ -0,0 +1,65 @@
+package seedu.address.model.pet.comparator;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.Comparator;
+
+import seedu.address.model.attendance.AttendanceEntry;
+import seedu.address.model.attendance.PresentAttendanceEntry;
+import seedu.address.model.pet.Pet;
+
+/**
+ * Comparator class to compare pets based on pet's Drop Off Time.
+ */
+public class PetDropOffTimeComparator implements Comparator<Pet> {
+    private LocalDate today = LocalDate.now();
+
+    /**
+     * Compares pet objects based on their drop off time attribute.
+     * @param first Pet to be compared with.
+     * @param second Pet to be compared with.
+     * @return Value signifying in the difference between the comparing attribute.
+     */
+    @Override
+    public int compare(Pet first, Pet second) {
+        AttendanceEntry firstAttendanceEntry = first.getAttendanceHashMap().getAttendanceEntry(today);
+        AttendanceEntry secondAttendanceEntry = second.getAttendanceHashMap().getAttendanceEntry(today);
+
+        boolean isFirstPresent = firstAttendanceEntry instanceof PresentAttendanceEntry;
+        boolean isSecondPresent = secondAttendanceEntry instanceof PresentAttendanceEntry;
```

Comment:
For these checks, maybe use `hasTransportArrangement()` that is defined in all `AttendanceEntry` classes instead of using `instanceof`?

Same for `PetPickUpTimeComparator`!

Reply:
Done!